src/lib/device-types/build-info-facade.ts

```diff
@@ -14,6 +15,33 @@ export const getIsIgnored = memoizee(
 	{ promise: true, preFetch: true, maxAge: BUILD_PROPERTY_CACHE_EXPIRATION },
 );
 
+export const getLogoUrl = memoizee(
+	async (
+		normalizedSlug: string,
+		buildId: string,
+	): Promise<string | undefined> => {
+		if (!FILES_HOST) {
+			return;
+		}
+		try {
+			const logoKey = getImageKey(normalizedSlug, buildId, 'logo.svg');
+			const hasLogo = await fileExists(logoKey);
+			if (!hasLogo) {
+				return;
+			}
+
+			// url encode since the buildId can contain a `+`
+			const encodedLogoPath = getImageKey(
+				...[normalizedSlug, buildId, 'logo.svg'].map(encodeURIComponent),
```

Comment:
At the start of the function:

```
const pathComponents = [normalizedSlug, buildId, 'logo.svg']
```

and then:

```suggestion
				...pathComponents.map(encodeURIComponent),
```

And also above:

```
const logoKey = getImageKey(... pathComponents);
```